lib/models/ObjectMD.js

```diff
@@ -1,23 +1,66 @@
+const constants = require('../constants');
+const VersionIDUtils = require('../versioning/VersionID');
 
-// Version 2 changes the format of the data location property
-// Version 3 adds the dataStoreName attribute
-const modelVersion = 3;
+const ObjectMDLocation = require('./ObjectMDLocation');
 
 /**
  * Class to manage metadata object for regular s3 objects (instead of
  * mpuPart metadata for example)
  */
-module.exports = class ObjectMD {
+class ObjectMD {
 
     /**
+     * Create a new instance of ObjectMD. Parameter <tt>objMd</tt> is
+     * reserved for internal use, users should call
+     * {@link ObjectMD.createFromBlob()} to load from a stored
+     * metadata blob and check the returned value for errors.
+     *
      * @constructor
+     * @param {ObjectMD|object} [objMd] - object metadata source,
+     *   either an ObjectMD instance or a native JS object parsed from
+     *   JSON
+     */
+    constructor(objMd = undefined) {
+        this._initMd();
+        if (objMd !== undefined) {
+            if (objMd instanceof ObjectMD) {
+                this._updateFromObjectMD(objMd);
+            } else {
+                this._updateFromParsedJSON(objMd);
+            }
+        } else {
+            // set newly-created object md modified time to current time
+            this._data['last-modified'] = new Date().toJSON();
+        }
+        // set latest md model version now that we ensured
+        // backward-compat conversion
+        this._data['md-model-version'] = constants.mdModelVersion;
```

Comment:
This is the constructor :wink: but yeah will do for getters/setters as you suggested :)

Reply:
Googling a bit tells me you can return something from a constructor, which is what will be returned by the `new` operator. So in our case I think we should not return explicitly a value because we are simply initializing the object that will be returned implicitly. Is this ok?

Reply:
Usually constructors return their internal state so that people do stuff like this
```
const location = new Foo().getLocation();
```

Reply:
I realize that this is an old habit. Constructors return `this` by default if they don't return anything. For better understanding we started down the path of `return this;` in constructors.